lib/zip/crypto/aes_encryption.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'openssl'
+
+module Zip
+  module AESEncryption # :nodoc:
+    VERIFIER_LENGTH = 2
+    BLOCK_SIZE = 16
+    AUTHENTICATION_CODE_LENGTH = 10
+
+    def initialize(password, strength)
+      @password = password
+      @strength = strength
+
+      n = @strength + 1
+      @headers = {
+        bits: 64 * n,
+        key_length: 8 * n,
+        mac_length: 8 * n,
+        salt_length: 4 * n
+      }
+      @counter = 0
+    end
+
+    def header_bytesize
+      @headers[:salt_length] + VERIFIER_LENGTH
+    end
+
+    def gp_flags
+      0
```

Comment:
As described below, the general purpose flag should be set to 1:
>  As for any encrypted file, bit 0 of the "general purpose bit flags" field must be set to 1 in each AES-encrypted file's local header and central directory entry.
- https://www.winzip.com/en/support/aes-encryption/
```suggestion
      1
```